Train Activity

prastut opened the pull request.

prastut committed:
WIP

prastut committed:
WIP

prastut committed:
WIP

prastut committed:
WIP

prastut committed:
WIP

prastut committed:
WIP

prastut committed:
Merge branch 'develop' into feature-activity-train

prastut committed:
WIP

prastut committed:
Init

prastut committed:
WIP

prastut committed:
WIP all activities complete

prastut committed:
All activities roughly done, orchestration left

prastut committed:
react cursor for nomalization of cordinates

prastut committed:
WIP

prastut committed:
merge develop, fix conflicts

houshuang committed:
Merge branch 'develop' into feature-activity-train

prastut committed:
WIP

prastut committed:
WIP

prastut committed:
Merge branch 'feature-activity-train' of https://github.com/chili-epfl/FROG into feature-activity-train

prastut committed:
Polish to Command

prastut committed:
WIP

prastut committed:
Polish only on dragDrop left

prastut committed:
WIP

prastut committed:
Drag&Drop polishing left

prastut committed:
Done

prastut committed:
Reset to all

jkolsen reviewed (changes requested):
- [x] Could you address the errors found in the tests that failed?

jkolsen commented on ac/ac-train/src/index.js:
```diff
@@ -0,0 +1,58 @@
+const config = {
+  type: 'object',
+  required: ['timeOfEachInstance'],
+  properties: {
```
Could we specify here if it should be in seconds or milliseconds?

prastut replied:
^ @lfaucon how would I do this?

jkolsen reviewed (dismissed):
Comments from walking through interface:

Buy and Help Button: 
- [x] Could we put the buy button and maybe one for the help? It is not clear that these are clickable. 
- [x] Could we put the help button by the buy button since they are both interactive and the timer is 

Rest: 
- [x] The graphical interface map was not working.
- [x] One the map, could we make the dot for Bern smaller?
- [x] Add in correctness feedback. 
- [x] Instead of "Activity 1"...etc. could we say "Interface 1"?
- [x] When the timer is double digits, it is slightly cut off by the help button.
not?
- [x] For the graphical, the map gets very small making it hard to use while the text stays the same size. It may be better to also shrink the text. (optional)

prastut committed:
fixing flow

prastut commented:
@jkolsen @lfaucon  which 🥇 ?

![screen shot 2018-04-12 at 2 43 44 pm](https://user-images.githubusercontent.com/10279686/38678418-419be0cc-3e61-11e8-8a39-b20b41509e80.png)
![screen shot 2018-04-12 at 2 47 03 pm](https://user-images.githubusercontent.com/10279686/38678419-41baf052-3e61-11e8-8f06-5bbff7ee3351.png)
![screen shot 2018-04-12 at 2 51 46 pm](https://user-images.githubusercontent.com/10279686/38678420-41d5f4b0-3e61-11e8-9412-67fdf47c5d3b.png)

jkolsen commented:
I vote for the first or third.

prastut committed:
all changes except feedback

prastut committed:
instanceCount and TimeofEachInstance becomes dynamic

prastut committed:
Merge branch 'develop' into feature-activity-train

prastut commented:
#985 is essential to get some of the errors fixed that flow throws. #986 is essential because it involves some breaking changes with MUI.

prastut committed:
Validation done

prastut committed:
Merge branch 'develop' into feature-activity-train

prastut committed:
added validation

prastut committed:
mui upgrade bugs fixed

prastut committed:
fixing build

prastut committed:
merge develop

prastut committed:
Fix merge

prastut committed:
Fix merge error

prastut committed:
dragDrop to recompose

prastut committed:
WIP

prastut committed:
Merge branch 'develop' into feature-activity-train

prastut committed:
WIP

prastut committed:
WIP

prastut committed:
WIP

prastut committed:
All dashboards done, polish left

jkolsen commented:
It looks like the progress dashboard still needs to be connected to the train activity.

jkolsen commented:
1. The interface currently only gives one type of ticket purchase (Geneva to lausanne) 
2. All interfaces are marking my input as wrong even when it is correct
3. The dashboards are putting all of the information under command whether I am filling out form or command etc

jkolsen commented:
The drag and drop activity just loaded as a hello world and crashed the dashboards

prastut commented:
@jkolsen apologies for the mess. I had to rewrite many files to get the dashboards working and forgot to remove the testing code. 

Give me some time to whip the code in shape.

prastut committed:
Safe point

prastut committed:
i think it works

prastut committed:
reformatting

prastut committed:
Done

prastut committed:
Progress dashboard

prastut committed:
Polishing

prastut committed:
working

prastut commented:
@jkolsen 

 > It looks like the progress dashboard still needs to be connected to the train activity.  
   
Fixed

> 1. The interface currently only gives one type of ticket purchase (Geneva to lausanne) 

Fixed

> 2. All interfaces are marking my input as wrong even when it is correct

Fixed. Needs to be checked. 

> 3. The dashboards are putting all of the information under command whether I am filling out form or command etc

Fixed.

prastut commented:
Documenting remaining thing: 

- [x] Problems with `time` reactive data structure. To check, for the very first interface -> first iteration, don't do anything, let the timer run out -> dashboards crash. Possibly will get fixed with the new logger. 

- [x] What if the student fills something out and doesn't press buy, should we make the answer undefined or use whatever state the answer is it in. Currently it's being undefined and so the answer is marked as false. 

- [x] Dragdrop crashes on preview because of conflict with React Mosaic react-dnd. https://github.com/react-dnd/react-dnd/issues/186

jkolsen commented on ac/ac-train/src/Dashboards/MeanPerInterface.jsx:
```diff
@@ -0,0 +1,126 @@
+    const coordinates = interfaces.map(int => {
+      if (whichDash === 'help') {
+        let countSum = 0;
+
```
When asking for help before any data has been put in the dashboards (help on the first instance of the first interface) the dashboards crash with error: cannot read property 0 of undefined.

jkolsen commented on ac/ac-train/src/ActivityRunner.jsx:
```diff
@@ -0,0 +1,107 @@
+  };
+
+  componentWillMount() {
+    this.interfaces = ['start', ...shuffle(['graphical', 'command', 'form'])];
```
Drag and drop still is not included as an interface

jkolsen commented:
@prastut 

The graphical interface is still marking everything as not correct.

jkolsen commented:
@prastut 

-  What if the student fills something out and doesn't press buy, should we make the answer undefined or use whatever state the answer is it in. Currently it's being undefined and so the answer is marked as false.

To the question above. Let's leave it as undefined. If they did not press buy then it is timing out (as real interfaces can) and they did not successfully buy. To purchase a ticket, they must press buy.

prastut committed:
Bandages

prastut committed:
bandages part 2

prastut committed:
bandages part 3

prastut commented:
@jkolsen 

> When asking for help before any data has been put in the dashboards (help on the first instance of the first interface) the dashboards crash with error: cannot read property 0 of undefined.

Fixed using a bandage. When we have the new logger, these problems of undefined will go away. 

> Drag and drop still is not included as an interface

Fixed. But will not work in preview. Works normally. 

> The graphical interface is still marking everything as not correct.

Fixed. 

> If they did not press buy then it is timing out (as real interfaces can) and they did not successfully buy. To purchase a ticket, they must press buy.

👍

prastut committed:
fix flow errors

prastut committed:
Drag and drop works in preview

jkolsen commented on ac/ac-train/src/Dashboards/MeanPerInterface.jsx:
```diff
@@ -0,0 +1,135 @@
+            </VictoryChart>
+          </Grid>
+        </Grid>
+      </Paper>
```
The legend for the time and error per interface plots have the legend in the middle of the plot.
<img width="536" alt="screen shot 2018-04-18 at 11 35 44 am" src="https://user-images.githubusercontent.com/2493446/38924380-1bdb9d8c-42fd-11e8-8eef-6548fa194373.png">

jkolsen commented:
On the intro screen, currently the interface type is not appearing. 
<img width="534" alt="screen shot 2018-04-18 at 11 35 49 am" src="https://user-images.githubusercontent.com/2493446/38924432-3df065ba-42fd-11e8-8734-97084fbfd6e8.png">

prastut commented:
Damn, I forgot to add the interfaces back. Just add graphical, form,
command to this.interfaces.

On Wed, Apr 18, 2018, 11:40 AM jkolsen <notifications@github.com> wrote:

> On the intro screen, currently the interface type is not appearing.
> [image: screen shot 2018-04-18 at 11 35 49 am]
> <https://user-images.githubusercontent.com/2493446/38924432-3df065ba-42fd-11e8-8734-97084fbfd6e8.png>
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/chili-epfl/FROG/pull/980#issuecomment-382328083>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AJzbBsh09MeuP-ZkHQ6_nansRyHp-LKSks5tpwn6gaJpZM4TPpxA>
> .
>

jkolsen commented on ac/ac-train/src/Interface.jsx:
```diff
@@ -0,0 +1,191 @@
+    case 'graphical':
+      return <Graphical {...props} />;
+    default:
+      return <h1>Hello World</h1>;
```
This is getting returned as one of the interfaces
<img width="534" alt="screen shot 2018-04-18 at 11 35 49 am" src="https://user-images.githubusercontent.com/2493446/38924979-a34318da-42fe-11e8-9c59-4181862ffcd9.png">

prastut commented:
https://github.com/chili-epfl/FROG/blob/6e834d8dc91647dfa50b801d43b1d28440a66e56/ac/ac-train/src/ActivityRunner.jsx#L66



You can add all the interfaces back to show all interfaces here.


On Wed, Apr 18, 2018, 11:50 AM jkolsen <notifications@github.com> wrote:

> *@jkolsen* commented on this pull request.
> ------------------------------
>
> In ac/ac-train/src/Interface.jsx
> <https://github.com/chili-epfl/FROG/pull/980#discussion_r182370624>:
>
> > +import { Form, Command, DragDrop, Graphical } from './Interfaces';
> +import Validation from './Validation';
> +import { getCommandForTicket, generateTicket } from './ActivityUtils';
> +
> +const RunActivity = props => {
> +  switch (props.activity) {
> +    case 'command':
> +      return <Command {...props} />;
> +    case 'form':
> +      return <Form {...props} />;
> +    case 'dragdrop':
> +      return <DragDrop {...props} />;
> +    case 'graphical':
> +      return <Graphical {...props} />;
> +    default:
> +      return <h1>Hello World</h1>;
>
> This is getting returned as one of the interfaces
> [image: screen shot 2018-04-18 at 11 35 49 am]
> <https://user-images.githubusercontent.com/2493446/38924979-a34318da-42fe-11e8-9c59-4181862ffcd9.png>
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/chili-epfl/FROG/pull/980#pullrequestreview-113150001>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AJzbBhoVy4UDZMhEYHF--qHw6xLqodFGks5tpwxRgaJpZM4TPpxA>
> .
>

jkolsen commented:
When I load the train activity (at least in preview), I get this error:

Stateless function components cannot be given refs. Attempts to access this ref will fail.
Check the render method of `DragDropContext(Content)`.

jkolsen commented on ac/ac-train/src/Guidelines.jsx:
```diff
@@ -0,0 +1,158 @@
+            Interface {step}
+          </Typography>
+        )}
+      </Grid>
```
We do not need to label the interface number that they are on, so this can probably be deleted.

prastut replied:
Removing the number looks weird: 

![image](https://user-images.githubusercontent.com/10279686/38929167-c1c89dc2-430b-11e8-9116-f174dc816a48.png)

prastut committed:
Fixed warnings

jkolsen commented:
For the first four graphs, there is always one extra point at the end of the graph that has not occurred. Could this point be deleted?
<img width="519" alt="screen shot 2018-04-18 at 12 51 55 pm" src="https://user-images.githubusercontent.com/2493446/38927900-7b8ad93c-4307-11e8-8199-dfa736b27587.png">

Commit:
Changes to train UI

prastut committed:
fix graph issues

prastut committed:
fix

Commit:
Moved legend

prastut committed:
fix for extra points in graph

prastut committed:
Merging Jenny's code

lfaucon committed:
Merge branch 'develop' into feature-activity-train

lfaucon committed:
Update Dashboard to new API (clean and easy)

lfaucon committed:
Eslint & Flow :)

lfaucon commented:
@jkolsen @prastut Great work on this activity. The dashboard code was so well written that it took me 2 minutes to update it to the new API. I haven't looked at any other parts of the code. I could make a few comments to improve the UI of each interface, but I think the best it to show the interface as it is now to Pierre tomorrow.
